app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,19 @@
+class LikesController < ApplicationController
+  before_action :like_params, only: [:create]
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:post_id])
+    @like = Like.new(author_id: @user.id, post_id: @post.id)
+    if @like.save
+      flash[:notice] = 'Like was successfully created.'
+    else
+      flash[:alert] = 'Failed to add like.'
+    end
+    redirect_to user_post_path(@user, @post)
+  end
```

Comment:
- [ ] As required for the project a user to be able to add a like to posts, kindly ensure that everything is working as expected. Right now, your application is showing this error while trying to add a new like.
![image](https://github.com/ANTOINE1128/Blog-App1/assets/83042742/5a117b77-f96c-4094-b018-76c2f02aea18)